Hyp3 Deployment ReadMe Edits (#2999)
ASFHyP3/hyp3: README.md

To allow HTTPS connections, HyP3 needs an SSL certificate that is valid for its deployment domain name (URL):

AWS console -> AWS Certificate Manager -> Request certificate:\
1. Select "Request a public certificate"
2. Click the orange "Next" button
3. Choose a "Fully qualified domain name". Domain name should be something like `hyp3-foobar.asf.alaska.edu` or for a test deployment `hyp3-foobar-test.asf.alaska.edu`.
3. Choose "DNS validation"
4. Copy the "CNAME name" and "CNAME value"

Then copy past the add the validation record to a row of
https://gitlab.asf.alaska.edu/operations/puppet/-/edit/production/modules/legacy_dns/files/asf.alaska.edu.db

> in the format `<CNAME_name> in CNAME <CNAME_value>.`  (see previous records for examples).

Reviewer: ```suggestion
Then create a validation record in
https://gitlab.asf.alaska.edu/operations/puppet/-/edit/production/modules/legacy_dns/files/asf.alaska.edu.db
of the form `<CNAME_name> in CNAME <CNAME_value>`, stripping the `.asf.alaska.edu` from the CNAME name  (see previous records for examples).
```

    - `CLOUDFORMATION_ROLE_ARN` (ASF only) - part of the `hyp3-ci` stack that you deployed, e.g. `arn:aws:iam::xxxxxxxxxxxx:role/hyp3-ci-CloudformationDeploymentRole-XXXXXXXXXXXXX`
    - `SECRET_ARN` - ARN for the AWS Secrets Manager Secret that you created manually, e.g. `arn:aws:secretsmanager:us-west-X:XXXXXXXXXXXX:secret:hyp3-foobar-XXXXXX`
    - `V2_AWS_ACCESS_KEY_ID` - AWS access key ID:
      - ASF: for the `github-actions` user (created in step "Enable CI/CD above")
      - JPL: for the service user
      - EDC: created by an ASF developer via Kion
    - `V2_AWS_SECRET_ACCESS_KEY` - The corresponding secret access key
    - `VPC_ID` - ID of the default VPC for this AWS account and region (aws console -> VPC -> Your VPCs, e.g. `vpc-xxxxxxxxxxxxxxxxx`)
    - `SUBNET_IDS` - Comma delimited list (no spaces) of the default subnets for the VPC specified in `VPC_ID` (aws console -> VPC -> Subnets, e.g. `subnet-xxxxxxxxxxxxxxxxx,subnet-xxxxxxxxxxxxxxxxx,subnet-xxxxxxxxxxxxxxxxx,subnet-xxxxxxxxxxxxxxxxx`)

### Create the HyP3 deployment

> You will need to add the deployment to the matrix in an existing GitHub Actions `deploy-*.yml` workflow (.github/workflows/) or create

Reviewer: ```suggestion
You will need to add the deployment to the matrix in an existing GitHub Actions `deploy-*.yml` workflow located in the `.github/workflows/` directory, or create
```

a new one for the deployment. If you need to create a new one, we recommend copying one of the

> existing workflows, and then updating all of the fields.

Reviewer: ```suggestion
existing workflows, and then updating all of the fields
```

>    ```
> Remember to remove this after the DNS PR is merged!

#### Testing and adding user credits to your hyp3 deployment

> After successful deployment and DNS record merging you can test you deployment by accessing the Swagger UI and using the POST /user tab to 

Reviewer: ```suggestion
After successfully deploying HyP3 and your new DNS record has taken effect (or you've edited your local DNS name resolution), you can test your deployment by accessing the Swagger UI and using the POST `/user` tab to 
```

check if your user is approved and has credits for running jobs on the deployment. This should automatically add your user to the DynamoDB 
table. 

AWS console -> DynamoDB -> Explore items
1. Find the table with format like `hyp3-foobar-UsersTable-XXXXXXXXXXXXX`
2. Edit your user (only present after using the Swagger UI in some way)
3. Add credits if neccessary

> You can then return the Swagger UI and use the POST /jobs to run a test job and confirm it completes.

Reviewer: ```suggestion
You can then return the Swagger UI and use the POST `/jobs` to run a test job and confirm it completes.
```
